### PR TITLE
Fixes #12606: Validates repo ids on enable/disable -r commands

### PR DESCRIPTION
**Pull request by sean797**
I know this is going to be replaced with a "inter satellite link" at some point but I made this change to my install as i kept thinking I had enabled a repo but after a configure the repo wasn't listed.

### CODE REVIEW

File: lib/disconnected_pulp.rb [hunk 1]
```diff
@@ -74,13 +74,20 @@ def clean
       end
     end
   end
-
+
```
Comment (ehelms) [inline lib/disconnected_pulp.rb]: Is this change needed?

File: lib/disconnected_pulp.rb [hunk 2]
```diff
@@ -74,13 +74,20 @@ def clean
...
-
+  
   def enable(value, repoids = nil, all = nil)
+    allrepoids = manifest.repositories.keys
     if repoids
       repoids = repoids.split(/,\s*/).collect(&:strip)
+      # Check all given repos are valid
+      repoids.each do |repoid|
+        unless allrepoids.include?(repoid)
+          LOG.fatal _("%{repoid} isn't listed in the imported manifest, see katello-disconnected list --disabled") % {:repoid => repoid}
```
Comment (ehelms) [inline lib/disconnected_pulp.rb]: I think this would be OK being just an error and not fatal.

**Comment by sean797** [comment]
Didn't notice the white space to be honest, also I've changed it to error instead of fatal - either way works. Are you happy for me to squash it into one commit?

**Comment by ehelms** [comment]
Yep -- if we want to get this in Katello 2.4 release, we'll need a Redmine issue (http://projects.theforeman.org/projects/katello/issues/new) and ideally the commit message linked to it.

**Comment by sean797** [comment]
Done :)

**Comment by ehelms** [comment]
Awesome, thanks @sean797 ! We'll just need to open a katello-packaging PR to bump the version after this.
